superset-frontend/src/pages/SavedQueryList/index.tsx

```diff
@@ -497,28 +489,22 @@ function SavedQueryList({
         ),
         paginate: true,
       },
-
-      {
-        Header: t('Search'),
-        id: 'label',
-        key: 'search',
-        input: 'search',
-        operator: FilterOperator.allText,
-      },
+      ...(isFeatureEnabled(FeatureFlag.TAGGING_SYSTEM) && canReadTag
+        ? [
+            {
+              Header: t('Tag'),
+              id: 'tags',
+              key: 'tags',
+              input: 'select',
+              operator: FilterOperator.savedQueryTags,
+              fetchSelects: loadTags,
+            },
+          ]
+        : []),
     ],
     [addDangerToast],
   );
 
-  if (isFeatureEnabled(FeatureFlag.TAGGING_SYSTEM) && canReadTag) {
-    filters.push({
-      Header: t('Tags'),
-      id: 'tags',
-      key: 'tags',
-      input: 'search',
-      operator: FilterOperator.savedQueryTags,
-    });
-  }
-
```

Comment:
This was done outside the useMemo, causing the tags filter to get appended multiple times to the filter array: 
<img width="1163" alt="image" src="https://github.com/apache/superset/assets/33317356/3734710a-5b98-45eb-9f76-de64789e36d3">